opcua/client/client.py

```diff
@@ -206,36 +207,42 @@ def connect_and_get_server_endpoints(self):
         """
         Connect, ask server for endpoints, and disconnect
         """
-        self.connect_socket()
-        self.send_hello()
-        self.open_secure_channel()
-        endpoints = self.get_endpoints()
-        self.close_secure_channel()
-        self.disconnect_socket()
+        try:
+            self.connect_socket()
+            self.send_hello()
+            self.open_secure_channel()
+            endpoints = self.get_endpoints()
+            self.close_secure_channel()
+        finally:
+            self.disconnect_socket()
         return endpoints
 
     def connect_and_find_servers(self):
         """
         Connect, ask server for a list of known servers, and disconnect
         """
-        self.connect_socket()
-        self.send_hello()
-        self.open_secure_channel()  # spec says it should not be necessary to open channel
-        servers = self.find_servers()
-        self.close_secure_channel()
-        self.disconnect_socket()
+        try:
+            self.connect_socket()
```

Comment:
Not sure if that one should be before or after try: ..... Before i guess. If it fails then no socket is created and no need to close it